### PR TITLE
IGVF-1517-schema-doc-update

### PR DESCRIPTION
**Pull request by jenjou**

### CODE REVIEW

**Review by gabdank — DISMISSED** [review]

File: docs/updating_schemas.md [hunk 1]
```diff
@@ -236,6 +324,75 @@ To add an object with accession prefix 'SM':
...
+    * `type`: Required. `string`, `number`, `integer`, `boolean`, `array`, or `object`.
+    * `description`: Required. A definition of the property.
+    * `default`: A default value automatically assigned if no value is provided upon submission.
+    * Strings:
+        * `enum`: A list of valid values when a property requires a controlled vocabulary.
+        * `linkTo`: A list of valid objects to link to. Must be the name of a schema, which can be abstract or concrete.
+        * `pattern`: A regex pattern applied to the string.
+    * Numbers or integers:
+        * `minimum`: A minimum value.
+        * `maximum`: A minimum value.
```
Comment (khzlin) [inline docs/updating_schemas.md:347]: change to `A maximum value`

File: docs/updating_schemas.md [hunk 2]
```diff
@@ -236,6 +324,75 @@ To add an object with accession prefix 'SM':
...
+
+
+### Stylebook
+The following table provides common types of schema changes and provides a specific sentence pattern to follow for the entry.
+|Type of schema change|Sentence example|
+|-|-|
+|Add a property|Add \[property name\].|
+|Remove a property|Remove \[property name\].|
+|Replace a property|Rename \[old name\] to \[new name\].|
+|Add an enum|Extend \[property name\] enum list to include \[enum\].|
```
Comment (khzlin) [inline docs/updating_schemas.md:386]: are there two options for this?

File: docs/updating_schemas.md [hunk 3]
```diff
@@ -236,6 +324,75 @@ To add an object with accession prefix 'SM':
...
+        * `items`: Required. A definition of the items that are valid in the array. Must include `type` at minimum.
+        * `minItems`: Required. An integer indicating the minimum number of items in an array. Must be at least 1.
+        * `maxItems`: An integer indicating the maximum number of items in an array.
+        * `uniqueItems`: Required. A boolean which is always `True`.
+    * Objects:
+        * `properties`: A nested schema. Must obey above requirements for each property: `title`, `type`, and `description`.
+
+The following are properties categorized by IGVF members' permission to submit. These properties must follow the preceding guidelines according to their `type` in addition to the guidelines below.
+* Submittable properties:
+    * `comment`: Required. Instructions for submitters explaining details of how to submit the property. If there is a dependency defined for the property, the comment must include: "This property is affected by dependencies. See \[link\]."
```
Comment (khzlin) [inline docs/updating_schemas.md:358]: not required, optional

File: docs/updating_schemas.md [hunk 4]
```diff
@@ -236,6 +324,75 @@ To add an object with accession prefix 'SM':
...
+    * `submissionExample`: Required. An example of how to specify the property using appscript and igvf_utils tools.
+* Admin only properties:
+    * `permission`: Required. A string property which is always `import_items`. Must not be present for any property that can be modified by any submitter.
```
Comment (khzlin) [inline docs/updating_schemas.md:361]: not required, optional

File: docs/updating_schemas.md [hunk 5]
```diff
@@ -236,6 +324,75 @@ To add an object with accession prefix 'SM':
...
+* Calculated properties:
+    * `notSubmittable`: Required. A boolean property which is always `True`. Must not be present for any submittable property.
```
Comment (khzlin) [inline docs/updating_schemas.md:363]: not required, optional

File: docs/updating_schemas.md (hunk 4 above)
Comment (jenjou) [reply docs/updating_schemas.md:361]: I put required here, but the intent was it should only apply to "admin only" properties. Is there a better way to make that more clear?

File: docs/updating_schemas.md (hunk 5 above)
Comment (jenjou) [reply docs/updating_schemas.md:363]: This is similar to admin only, it is "required" but only for calculated properties, is there a way to make it more clear?

File: docs/updating_schemas.md (hunk 2 above)
Comment (jenjou) [reply docs/updating_schemas.md:386]: I'm a little confused on this comment, I think we only discussed one option for "adding an enum". But maybe I am forgetting?

Comment (khzlin) [reply docs/updating_schemas.md:386]: option 1) Adding an envm vs. 2) Extend enum list to include...

File: docs/updating_schemas.md (hunk 4 above)
Comment (khzlin) [reply docs/updating_schemas.md:361]: ohh i didn't see the indent. I think that's fine.

File: docs/updating_schemas.md (hunk 5 above)
Comment (khzlin) [reply docs/updating_schemas.md:363]: ohh i didn't see the indent. I think that's fine.

File: docs/updating_schemas.md (hunk 2 above)
Comment (khzlin) [reply docs/updating_schemas.md:386]: this is resolved. This is a table format and I was reading this as `option1|option2`

**Pull request merged**